R/utils.R

```diff
@@ -129,31 +129,65 @@ render_dependencies <- function(dependencies, local = TRUE) {
     } else {
       "file"
     }
-    htmltools::renderDependencies(list(dep), src, encodeFunc = identity)
+    
+    # According to Dash convention, label react and react-dom as originating
+    # in dash_renderer package, even though all three are currently served
+    # u p from the DashR package
+    if (dep$name %in% c("react", "react-dom")) {
+      dep$name <- "dash-renderer"
+    }
+    
+    # The following lines inject _dash-component-suites into the src tags,
+    # as this is the current Dash convention. The dependency paths cannot
+    # be set solely at component library generation time, since hosted
+    # applications should have the app name injected as well.
+    #
+    # This is essentially analogous to this codeblock on the Python side:
+    # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L207
+    #
+    # Use the system file modification timestamp for the current
+    # package and add the version number of the package as a query
+    # parameter for cache busting
+    
+    if (!is.null(dep$package)) {
+      # the gsub line is to remove stray duplicate slashes, to
+      # permit exact string matching on pathnames
+      dep_path <- gsub("//+",
+                       "/",
+                       paste(dep$src$file,
+                             dep$script,
+                             sep = "/")
```

Comment:
If this is only used as a file path, could we use `file.path`? I recognize that it's being used below, but just to get into the habit of never constructing file paths by hand.

(unless this variable is also used as a script tag / url pathname? By reading the code below it didn't seem like it but I might've missed something)

Reply:
fixed in 3a61d023c1603b9d6d1f6303351b620cbb5262ac